lib/rubygems/specification.rb

```diff
@@ -763,6 +763,13 @@ def test_files=(files) # :nodoc:
 
   attr_accessor :specification_version
 
+  ##
+  # The checksum for this gem
+  #
+  # Do not set this, it is set when the gemspec is indexed (if possible).
```

Comment:
if we don't want it to be set, should we make it an `attr_reader` and use a different method to write to the ivar?